src/components/BlogPage.js

```diff
@@ -1,49 +1,98 @@
 import axios from "axios";
 import { useEffect, useState } from "react";
 import { API_URL } from "./Api";
+import { FaRegEdit } from "react-icons/fa";
 
 const BlogPage = () => {
   const [displayBlogs, setDisplayBlogs] = useState([]);
+  const [page, setPage] = useState(1);
+  const [loading, setLoading] = useState(false);
+  const [noData, setNoData] = useState(false);
+
+  const getBlog = async (page) => {
+    try {
+      let url;
+      if ((page != null) && (page > 1)) {
+        url = `${API_URL}?page=${page}`;
+      } else {
+        url = API_URL;
+      }
+      const { data } = await axios.get(url);
+      setDisplayBlogs(data.getBlogPost.docs);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+  
+
+  window.onscroll = () => {
+    if (
+      window.innerHeight + document.documentElement.scrollTop ===
+      document.documentElement.offsetHeight
+    ) {
+      if (!noData) {
+        loadBlogs(page);
+      }
+    }
+  }
```

Comment:
Rewrite this without tying it to the `window` object.

Use event listeners in your `useEffect`